vision/product-search/cloud-client/src/main/java/com/example/vision/ProductInProductSetManagement.java

```diff
@@ -135,6 +144,50 @@ public static void removeProductFromProductSet(
   }
   // [END vision_product_search_remove_product_from_product_set]
 
+  // [START vision_product_search_purge_products_in_product_set]
+  /**
+   * Delete all products in a product set.
+   *
+   * @param projectId - Id of the project.
+   * @param location - Region name.
+   * @param productSetId - Id of the product set.
+   * @param force - Perform the purge only when force is set to True.
+   * @throws Exception - any error.
+   */
+  public static void purgeProductsInProductSet(
+          String projectId, String location, String productSetId, boolean force)
+          throws Exception {
+    try (ProductSearchClient client = ProductSearchClient.create()) {
+
+      String parent = LocationName.format(projectId, location);
+      ProductSetPurgeConfig productSetPurgeConfig = ProductSetPurgeConfig
+              .newBuilder()
+              .setProductSetId(productSetId)
+              .build();
+
+      PurgeProductsRequest req = PurgeProductsRequest
+              .newBuilder()
+              .setParent(parent)
+              .setProductSetPurgeConfig(productSetPurgeConfig)
+              // The operation is irreversible and removes multiple products.
+              // The user is required to pass in force=True to actually perform the
+              // purge.
+              // If force is not set to True, the service raises an exception.
+              .setForce(force)
+              .build();
+
+      //TODO: once its supported in all regions, will change it to 60 sec.
+      // testing method with region asia-east1 seems bit slower than normal.
```

Comment:
Can we delete the TODO? or are we waiting for a feature to launch?

If waiting? Please add `DO_NOT_MERGE` to PR title.

Reply:
For now  I added DO_NOT_MERGE label, I ran tests multiple times to see the average length of time to get LRO from response. It fluctuates between 40- 80sec. Therefore, I think it is good idea to keep as it is. [90sec].

Reply:
Yea, 90 seconds is fine, it's mostly just the TODO message that I don't think we want in a sample.